starlite/utils/sync.py

```diff
@@ -15,14 +18,28 @@
 )
 
 from anyio.to_thread import run_sync
-from typing_extensions import Literal, ParamSpec
-
-from starlite.utils.predicates import is_async_callable
+from typing_extensions import Literal, ParamSpec, TypeGuard
 
 P = ParamSpec("P")
 T = TypeVar("T")
 
 
+def is_async_callable(value: Callable[P, T]) -> TypeGuard[Callable[P, Awaitable[T]]]:
```

Comment:
I moved it to this file to resolve the cyclical import issues that were happening.

Reply:
I see. I guess it makes sense to have this there.